scripts/sign_rproc_fw.py

```diff
@@ -0,0 +1,568 @@
+#
+# Copyright (C) 2023, STMicroelectronics - All Rights Reserved
+#
+
+try:
+    from elftools.elf.elffile import ELFFile
+    from elftools.elf.sections import SymbolTableSection
+    from elftools.elf.enums import ENUM_P_TYPE_ARM
+    from elftools.elf.enums import *
+except ImportError:
+    print("""
+***
+ERROR: "pyelftools" python module is not installed or version < 0.25.
+***
+""")
+    raise
+
+try:
+    from Cryptodome.Hash import SHA256
+    from Cryptodome.Signature import pkcs1_15
+    from Cryptodome.PublicKey import RSA
+    from Cryptodome.Signature import DSS
+    from Cryptodome.PublicKey import ECC
+except ImportError:
+    print("""
+***
+ERROR: "pycryptodomex" python module should be installed.
+***
+""")
+    raise
+
+import os
+import sys
+import struct
+import logging
+import binascii
+
+#  Generated file structure:
+#
+#                    ----+-------------+
+#                   /    |    Magic    |
+#                  /     +-------------+
+#                 /      +-------------+
+#                /       |   version   |
+#               /        +-------------+
+#              /         +-------------+
+# +-----------+          | sign size   |   size of the signature
+# |   Header  |          +-------------+   (in bytes, 64-bit aligned)
+# +-----------+          +-------------+
+#              \         | images size |   size of the images to load
+#               \        +-------------+   (in bytes, 64-bit aligned)
+#                \       +-------------+
+#                 \      |  TLV size   |   Generic TLV size
+#                  \     +-------------+   (in bytes, 64-bit aligned)
+#                   \    +-------------+
+#                    \   | PLAT TLV sz |   Platform TLV size
+#                     \--+-------------+   (in bytes, 64-bit aligned)
+#                        +-------------+
+#                        |     TLV     |   Information used to authenticate the
+#                        |             |   firmware, stored in
+#                        +-------------+   Type-Length-Value format.
+#                        +-------------+
+#                        |   Platform  |   Specific platform information,
+#                        |     TLV     |   stored in Type-Length-Value format.
+#                        +-------------+
+#                        +-------------+   Signature of the header, the TLV,
+#                        | Signature   |   and the platform TLV.
+#                        +-------------+
+#
+#                        +-------------+
+#                        |   Firmware  |
+#                        |    image 1  |
+#                        +-------------+
+#                               ...
+#                        +-------------+
+#                        |   Firmware  |
+#                        |    image n  |
+#                        +-------------+
+#
+#  TLV and platform TLV chunk:
+#
+#                  ------+-------------+
+#                  |     |  TLV Magic  |
```

Comment:
I assume TLV is supposed to be short for Type, Length, and Value. Why not use those names for the fields too?

Reply:
Please could you clarify? What fields are you talking about? 
The TLV structures are stored in TLV payload, the TLV header is inspired from MCUBOOT

Reply:
You call this field `TLV Magic` instead of just `Type`, same below `TLV size` vs `Length` and so on.

Reply:
The `TLV magic ` is not the TLV, but part of the header of the chunk ( or blob). 
- the `TLV header` contains 2 fields:
  - `Magic`: a magic value  that correxpond to the ID of the chunk
  - `size` the size of the TLV payload it is 'n' * sizeof(TLV structure), n is the number of TLV structures added.
-  the ` TLV payload `contains the TLV structures. It can contain the 'n' TLV structures  
 
I  will add  more comments